Samples for Extending Pentaho/Reference Implementations/Security/SAML 2.0/pentaho-saml/src/main/java/org/pentaho/platform/spring/security/saml/PentahoSamlUserDetailsService.java

```diff
@@ -25,13 +27,14 @@
 
   private static final String PROVIDER_NAME = "providerName";
 
-  UserDetailsService userDetailsService;
-  String selectedAuthorizationProvider;
+  private UserDetailsService userDetailsService;
+  private String selectedAuthorizationProvider;
 
   // if the selectedAuthorizationProvider is our own ( i.e. samlId ), we'll directly use our own
-  UserDetailsService samlUserDetailsService;
-  String samlId;
-
+  private PentahoSamlNativeUserDetailsService samlUserDetailsService;
```

Comment:
PentahoSamlNativeUserDetailsService implements 2 interfaces AFAICT ( SAMLUserDetailsService, UserDetailsService ).

Why are we moving from working with the contracts / interfaces to working directly with that one implementing class PentahoSamlNativeUserDetailsService ? If we are relying on some class-specific methods, then we could be breaking the abstraction layer.

Is it possible to continue on working with the UserDetailsService interface ?